src/Entity.cpp

```diff
@@ -209,3 +209,30 @@ std::list<Entity*> Entity::get(std::filesystem::path& path) {
 void Entity::clear() {
   *this = Entity();
 }
+
+void Entity::delete_by_predicate(std::function<bool(const Entity&)> p){
```

Comment:
This function is a bit bloated but I did not see any direct way to make it more concise.

Reply:
Yes, not problem here @Bzero, just the way the `Entity` struct is set up.

Reply:
Something of an aside, but it makes me think that these separate members for each type of entity (`namespaces`, `groups`, `types`, etc) are no longer necessary really, they could all just be in one `children` and filtered out by type in `MarkdownGenerator` and other places, a bit like the way `MarkdownGenerator::view()` works. But something for another time.